### PR TITLE
Adds all dynamic carousel functionality, autocomplete for address input

### PR DESCRIPTION
**Pull request by smruthibalajee**
**Description:**

Carousels on main page are fully dynamic, including the thumbnail image. Retrieves name, thumbnail image (served as a blob key), and a hard-coded walking distance from Firestore. Query filters businesses that are displayed on each carousel by the tags (same functionality can be used for a primitive search). Additionally, businesses shown in the carousel are filtered by the city that the user's address is in. Address input field has autocomplete feature that uses the Places API to suggest locations as the user types in the text box. Uses Geocoding API to convert the entered address to latitude and longitude, then reverse geocodes to find the city. Additionally, created file to hardcode one business to Firestore; planning to add more and update the way that tags are stored.

**Key Features:**

- **Address autocomplete**: Uses Places API to implement autocomplete feature. When user clicks on one of the autocomplete suggestions, it constructs address, extracts city, and redirects them to the main page. Autocomplete is biased to US, will suggest locations within the country.

- **Address geocoding + reverse geocoding**: Uses Geocoding API to convert an address string to latitude and longitude coordinates that are stored to help with the user walking time calculations in the carousels (still in progress). The latitude and longitude are then reverse geocoded to get the city that the user inputted. 

- **Carousel query by tags and city**: The city extracted from user input is then compared with the city of every business in Firestore to only pull up businesses in the correct city. In order to populate the carousels, the tags of every business are checked to see if one matches the tag needed for the carousel (e.g: black-owned, trending, new, etc).

- **Dynamically loaded carousel content**: The carousel content is dynamically loaded by reconstructing the HTML elements within the script function and adding in the relevant fields from the business query. The card element is then added to the carousel element. 

- **Dynamically loaded thumbnail image**: The thumbnail image is stored as a blob key once the user uploads it to the webpage. This blob key is dynamically loaded from the Firestore database and served when the carousels are loaded.

**Key files to review:** main.html, script.js

**Functionality GIF's:**
autocomplete_dynamicmain

**Next steps:**
- Use distance matrix API to dynamically load walking time
- Add simple search to nav bar (no autocomplete)

### CODE REVIEW

File: ripple/target/portfolio-1/script.js [hunk 1]
```diff
@@ -23,97 +115,79 @@
...
+function geoQuery() {
+  //const GeoFirestore = geofirestore.initializeApp(db);
+  var GeoFirestore = new GeoFirestore(db);
+  // Create a GeoCollection reference
+  const geocollection = GeoFirestore.collection('businesses');
 
-/* Display an alert containing the inputted address if user presses enter */
-function searchAddress(e) {
-  addressInput = document.getElementById("address-input").value;
-  if (e.keyCode === 13) {
```
Comment (peter14f) [inline ripple/target/portfolio-1/script.js:28]: Define a ENTER_KEY to make this a little more readable.

File: ripple/src/main/webapp/script.js [hunk 2]
```diff
@@ -1,37 +1,169 @@
...
+          var carouselInner = carouselElement.getElementsByClassName('carousel-inner row w-100 mx-auto')[0];
+          alert(contentStrings.join("\n"));
+          carouselInner.innerHTML = contentStrings.join("\n");
+          alert(document.getElementById('card-dynamic-image').src);
+          carouselInner.firstElementChild.className += " active";
+          $(carouselElement).carousel({slide : true, interval : false});
+        }
+   });
+}
+
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/script.js:164]: Add landing page-specific and main page-specific functions either to landing.js or main.js logic. Script.js should contain functions that every page will use.

Comment (smruthibalajee) [reply ripple/src/main/webapp/script.js]: good point, has been updated

File: ripple/src/main/webapp/landing.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        //Convert address if user clicks on the autocomplete suggestion
+        convertAddressToCoord(address);
+        convertCoordToCity();
+        window.location.assign("main.html");
+    });
+}
+
+/* Display an alert containing the inputted address if user presses enter */
+function searchAddress(e) {
+  if (e.keyCode === 13) {
```
Comment (peter14f) [inline ripple/src/main/webapp/landing.js:78]: Define a ENTER_KEY to make this a little more readable.

Comment (smruthibalajee) [reply ripple/src/main/webapp/landing.js]: fixed!

File: ripple/src/main/webapp/landing.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        //Set the input address to use in the main page
+        localStorage.setItem('inputAddress', address);
+        window.location.assign("main.html");
+    });
+}
+
+/* Display an alert containing the inputted address if user presses enter */
+function searchAddress(keyPress) {
+  var ENTER_KEY = 13;
```
Comment (peter14f) [inline ripple/src/main/webapp/landing.js:38]: Probably makes more sense to be a global variable outside of the function.

Comment (smruthibalajee) [reply ripple/src/main/webapp/landing.js:38]: Ok, will change!

**Pull request merged**